compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/error.todo-for-loop-with-context-variable-iterator.expect.md

```diff
@@ -22,7 +22,7 @@ function Component() {
   4 |   // NOTE: `i` is a context variable because it's reassigned and also referenced
   5 |   // within a closure, the `onClick` handler of each item
 > 6 |   for (let i = MIN; i <= MAX; i += INCREMENT) {
-    |        ^^^^^^^^^^^ Todo: Support for loops where the index variable is a context variable. `i` is a context variable (6:6)
+    |                               ^ InvalidReact: Updating a value used previously in JSX is not allowed. Consider moving the mutation before the JSX. Found mutation of `i` (6:6)
```

Comment:
This is a false positive as we're not mutating but just reassigning to `i`. In this case, the lambda is guaranteed to only capture the correct instance of `i`.

Happy to fix before landing or in a followup. Note that this case now incorrectly bails out:

```js
// same code, just unrolling the loop once
function Component() {
  const data = useData();
  const items = [];
  let i = 0;
  items.push(<Stringify key={i} onClick={() => data.set(i)} />);
  i += 1;
  items.push(<Stringify key={i} onClick={() => data.set(i)} />);
  return items;
}
```